uaclient/entitlements/__init__.py

```diff
@@ -29,11 +32,22 @@ def cast(_, x):  # type: ignore
 )  # type: Dict[str, Type[UAEntitlement]]
 
 
-ALL_ENTITLEMENTS_STR = ", ".join(sorted(ENTITLEMENT_CLASS_BY_NAME.keys()))
-RELEASED_ENTITLEMENTS_STR = ", ".join(
-    [
-        k
-        for k, cls in sorted(ENTITLEMENT_CLASS_BY_NAME.items())
-        if not cls.is_beta
-    ]
-)  # type: str
+def valid_services(allow_beta: bool = False) -> "List[str]":
+    """Return a list of valid (non-beta) services.
+
+    @param allow_beta: if we should allow beta services to be marked as valid
+    """
+    cfg = UAConfig()
+    allow_beta_cfg = is_config_value_true(cfg.cfg, "features.allow_beta")
+    allow_beta |= allow_beta_cfg
+
+    if allow_beta:
+        return sorted(ENTITLEMENT_CLASS_BY_NAME.keys())
+
+    return sorted(
+        [
+            ent_name
+            for ent_name, ent_cls in ENTITLEMENT_CLASS_BY_NAME.items()
+            if not ent_cls.is_beta
+        ]
+    )
```

Comment:
I think now with the `valid_service` method on entitlements, this can be simplified to

```suggestion
    cfg = UAConfig()
    return sorted(
        [
            ent_name
            for ent_name, ent_cls in ENTITLEMENT_CLASS_BY_NAME.items()
            if ent_cls(cfg=cfg, allow_beta=allow_beta).valid_service
        ]
    )
```

Reply:
Yes, this a valid refactoring. My rationale to separate the two cases here is just to avoid creating the service objects if not needed. But maybe this optimization is not worth it at this point

Reply:
That makes sense - I think in this case my preference would be to favor readability over the small optimization from avoiding object instantiation - but that's not a strong preference here, feel free to ignore :)